### ISSUE TITLE
How to encrypt data so that another address can decrypt it?

### ISSUE THREAD

**Issue by RezaRahemtola** [aleph-im/aleph-sdk-ts#95]
Hello, I'm the current project manager of [InterPlanetaryCloud](https://github.com/PoCInnovation/InterPlanetaryCloud), which is based on this SDK.

**Is your feature request related to a problem? Please describe.**
IPC includes a drive with encrypted files. One of our key feature is file sharing but it will no longer work with the latest versions of the SDK :slightly_frowning_face: 

To share a file with someone, it currently works this way:
- A contact can be added by inputting his address (should be done for both parties)
- When a file is shared, its encryption key is itself encrypted with our contacts's public key (to which we no longer have access since #33 ) and our private key
- The receiver will then decrypt it with his own private key and the public key of the sender, allowing him to visualize the file content.

**Describe the solution you'd like**
Correct me if I'm wrong, but now it's not possible to do this (encrypt a content for someone else and retrieve a content encrypted for you), using your encrypt/decrypt methods the parts related to the private key can be handled, but not those related to the public key.

Do you have a functionality that I've missed for this use case, or is it in your roadmap?

**Describe alternatives you've considered**
Staying on the current SDK version (`2.2.2`), but that's not viable and we also plan on implementing account providers so we'll need the latest features on this part.

**Additional context**
- [Part of the code where we share a file](https://github.com/PoCInnovation/InterPlanetaryCloud/blob/master/src/lib/contact.ts#L293)
- [InterPlanetaryCloud demonstration](https://ipc.poc-innovation.com/)

**Comment by hoh** [comment]
Thank you for the clear issue report.

Allowing users to encrypt towards a third party address / public key should be supported when possible (some chains use a hash of the public key as address).

**Comment by BjrInt** [comment]
We're currently discussing this internally, and this feature will probably be integrated in the future. The main pain points are that not all chains handle keypairs the same way hence the removal a few months ago.

However, if you need to upgrade to a newer version of the SDK, [that property is exposed by the ethers Wallet](https://docs.ethers.io/v5/api/signer/#Wallet--properties), and a quick-fix solution could be to inherit the ETHAccount class directly :
```js
import { ETHAccount } from "aleph-sdk-ts/accounts/ethereum";
import ethers from "ethers";

export class ExposedPublicKey extends ETHAccount {
    public publicKey: string;

    constructor(wallet: ethers.Wallet, address: string) {
        super(wallet, address);
        this.publicKey = wallet.publicKey;
    }
}

// ...

const delegateETHAccount = new ExposedPublicKey(new ethers.Wallet(privateKey), wallet.address);
```

**Comment by BjrInt** [comment]
This is now a feature, (#104) and available starting from v3.20.
